server/src/main/java/io/crate/expression/scalar/ArrayUpperFunction.java

```diff
@@ -206,22 +205,17 @@ public Query toQuery(Function parent, Function arrayLength, Context context) {
                 if (cmpVal == 0) {
                     return new MatchNoDocsQuery("array_length([], 1) is NULL, so array_length([], 1) = 0 can't match");
                 }
-                return genericAndDocValueCount(parent, context, arrayRef, valueCountIsMatch);
+                return docValueCountOrGeneric(parent, context, arrayRef, valueCountIsMatch);
 
             case GtOperator.NAME:
-                if (cmpVal == 0) {
-                    return IsNullPredicate.refExistsQuery(arrayRef, context, false);
-                }
```

Comment:
I think idea of handling it separately was trying to improve performance of this case.

Do I get it right, that `IsNullPredicate.refExistsQuery `is itself buggy (as discovered recently) and the only way to fix it is to add `genericFunctionQuery` -> and hence it doesn't make sense to handle it separately anymore?

ie "proper, fixed in the future" `IsNullPredicate.refExistsQuery` can be as slow as `docValueCountOrGeneric` - are we 100% sure about that? Otherwise still makes sense to keep it, if it's still faster in some scenarios.